lib/fast_jsonapi/serialization_core.rb

```diff
@@ -15,56 +15,63 @@ class << self
                       :cache_length,
                       :cached
       end
-    end
 
-    class_methods do
-      def id_hash(id, record_type)
-        return { id: id.to_s, type: record_type } if id.present?
-      end
+      attr_accessor :object
 
-      def ids_hash(ids, record_type)
-        return ids.map { |id| id_hash(id, record_type) } if ids.respond_to? :map
-        id_hash(ids, record_type) # ids variable is just a single id here
+      def klass
+        self.class
       end
 
       def attributes_hash(record)
-        attributes_to_serialize.each_with_object({}) do |(key, method_name), attr_hash|
-          attr_hash[key] = record.public_send(method_name)
+        klass.attributes_to_serialize.each_with_object({}) do |(key, method_name), attr_hash|
+          attr_hash[key] = send(method_name)
         end
       end
 
       def relationships_hash(record, relationships = nil)
-        relationships = relationships_to_serialize if relationships.nil?
+        relationships = klass.relationships_to_serialize if relationships.nil?
 
         relationships.each_with_object({}) do |(_k, relationship), hash|
           name = relationship[:key]
           id_method_name = relationship[:id_method_name]
           record_type = relationship[:record_type]
           empty_case = relationship[:relationship_type] == :has_many ? [] : nil
           hash[name] = {
-            data: ids_hash(record.public_send(id_method_name), record_type) || empty_case
+            data: klass.ids_hash(record.public_send(id_method_name), record_type) || empty_case
           }
         end
       end
 
       def record_hash(record)
-        if cached
-          record_hash = Rails.cache.fetch(record.cache_key, expires_in: cache_length) do
-            temp_hash = id_hash(record.id, record_type) || { id: nil, type: record_type }
-            temp_hash[:attributes] = attributes_hash(record) if attributes_to_serialize.present?
+        self.object = record
```

Comment:
Consider making sure that `object` is only available within the context of this method, i.e.

```ruby
self.object = record
begin
  ...
ensure
  self.object = nil
end
```